src/panels/config/energy/components/ha-energy-device-settings.ts

```diff
@@ -39,6 +42,16 @@ export class EnergyDeviceSettings extends LitElement {
   @property({ attribute: false })
   public validationResult?: EnergyPreferencesValidation;
 
+  private _deviceKeys = new WeakMap<any, string>();
+
+  private _getKey(device) {
+    if (!this._deviceKeys.has(device)) {
+      this._deviceKeys.set(device, Math.random().toString());
```

Comment:
Isn't `device.stat_consumption` already unique and good enough for a key?

Reply:
You are right, got a little too copy-paste happy 😅